index.js

```diff
@@ -24,6 +24,35 @@ if (isWsl && which.sync('firefox', { nothrow: true })) {
   isWsl = false
 }
 
+/**
+ * Takes a string from Windows' tasklist.exe with the following arguments:
+ * `/FO CSV /NH /SVC` and returns an array of PIDs.
+ * @param {string} tasklist Expected to be in the form of:
+ * `'"firefox.exe","14972","Console","1","5.084 K"\r\n"firefox.exe","12204","Console","1","221.656 K"'`
+ * @returns {string[]} Array of String PIDs. Can be empty.
+ */
+ const extractPids = tasklist => tasklist
+    .split (',')
+    .filter (x => /^"\d{3,10}"$/.test (x))
+    .map (pid => pid.replace (/"/g, ''))
+
+ /**
+  * Curried function version of safeExecSync with reference to logger
+  * in a closure.
+  * @param {function} log An instance of logger.create
+  * @returns {{(command:string):string}}
+  */
+const createSafeExecSync = log => command => {
+  let output = ''
+  try {
+    output = String (execSync(command))
+  } catch (err) {
+    log.warn (String (err))
```

Comment:
Should this be `log.error` ?

Reply:
My thinking was that when multiple Firefox browsers are used then `taskkill.exe` might throw without it being an issue when multiple `taskkill.exe` commands are issued for PIDs that are already killed. But I just tested that and `taskkill.exe` does not throw in that case but simply returns:

```
ERROR: The process with PID 11464 could not be terminated.
Reason: There is no running instance of the task.
```

So I will change it to `log.error` :)

Reply:
Looking at your first log, it looks like you triggered an throwable error in `safeExecSync` without it being a _"real"_ error.

> 07 08 2021 11:01:06.390:WARN [FirefoxLauncher]: Error: Command failed: taskkill.exe /F /PID 2696 /PID 1788 /PID 28836 /PID 24164 /PID 12664 /PID 30600 /PID 31944 /PID 22560 /PID 3248 /PID 20676 /PID 13016 /PID 24816

In that case, I think it should only be a warning since the test will run as usual.

Reply:
I've changed it to `log.debug`... Since it's perfectly fine.. Probably.. Might be useful to see while debugging